go/libkb/merkle_client.go

```diff
@@ -1138,7 +1216,9 @@ func (mc *MerkleClient) verifySkipSequence(m MetaContext, ss SkipSequence, thisR
 		right = *lastRoot.Seqno()
 	}
 
-	if opts.historical && left < right {
+	// In historical queries (for which we fetch old roots), we check the skip
+	// sequence in the opposite direction.
+	if opts.historical {
```

Comment:
why is it safe to drop the second half of the conjunction?

Reply:
Because now historical queries are supposed to only go backwards (to a root which precedes your own), never forward. It is actually not safe to leave it there.